Source/WebCore/html/HTMLDialogElement.cpp

```diff
@@ -92,8 +92,10 @@ ExceptionOr<void> HTMLDialogElement::showModal()
 
     setIsModal(true);
 
-    if (!isInTopLayer())
+    if (!isInTopLayer()) {
+        updateContentRelevancyStatusIfNeeded(ContentRelevancyStatus::Visible, true);
         addToTopLayer();
+    }
```

Comment:
Since the spec says to make _all_ top layer elements relevant, I think you want to move this `updateContentRelevancyStatusIfNeeded(ContentRelevancyStatus::Visible, true);` call to `addToTopLayer()`. Also do we need a reverse call in `removeFromTopLayer()`? content can remain visible, but irrelevant after removing from top layer (fullscreen for instance).

Reply:
Good points, fixed.

Reply:
Would be nice to write WPT for popovers/fullscreen. I think the current ones only cover dialog?

Reply:
Great idea, I added the ones for popover:
https://github.com/web-platform-tests/wpt/pull/40541